src/VisualStudio/Core/Def/Packaging/PackageInstallerServiceFactory.cs

```diff
@@ -93,58 +102,32 @@ internal partial class PackageInstallerService : AbstractDelayStartedService, IP
             _packageSourceProvider = packageSourceProvider;
         }
 
-        public async ValueTask<ImmutableArray<PackageSource>?> TryGetPackageSourcesAsync(bool allowSwitchToMainThread, CancellationToken cancellationToken)
+        public ImmutableArray<PackageSource> TryGetPackageSources()
         {
-            // Only read from _packageSourcesAsync once, since OnSourceProviderSourcesChanged could reset it to default
-            // at any time while this method is running.
-            JoinableTask<ImmutableArray<PackageSource>?> packageSourcesAsync;
+            Task<ImmutableArray<PackageSource>> localPackageSourcesTask;
             lock (_gate)
             {
-                if (_packageSourcesAsync is null)
-                {
-                    _packageSourcesAsync = ThreadingContext.JoinableTaskFactory.RunAsync(() => GetPackageSourcesImplAsync());
-                }
+                if (_packageSourcesTask is null)
+                    _packageSourcesTask = Task.Run(() => GetPackageSourcesAsync());
```

Comment:
Yes.  Taht is the exact behavior *until* the point that the nuget team provides us with a safe async version of this API as well.  It's on their list.

1. you are correct, it's not a change in behavior.
2. it's not ideal behavior, but we accept it because we don't have a good alternative.  First, we can't deadlock :).  Second, we don't want to preemptively load this information as it can be costly during things like VS load.  So we just wait until we're asked the first time, understanding that it still may result ina  short time before it's available.

C'est la vie for the time being.

Reply:
Tagging @zivkan that this woudl be the next great API to ahve an async version of.  Thanks! :)

Reply:
Thanks for the suggestion: https://github.com/NuGet/Home/issues/9811